packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.js

```diff
@@ -4,18 +4,23 @@ const providerName = require('../../lib/constants').ProviderName;
 const configurationManager = require('../../lib/configuration-manager');
 
 class S3 {
-  constructor(context, options = {}) {
-    return (async () => {
-      let cred = {};
-      try {
-        cred = await configurationManager.loadConfiguration(context);
-      } catch (e) {
-        // ignore missing config
-      }
+  constructor(context, options = {}, test = false) {
```

Comment:
I don't have a whole lot of context around how this flag is going to be used, but in general "test" flags are a code smell. Could we just have the credentials be an input to the constructor and then it's up to the caller to supply real credentials or test credentials depending on the use case?